### ISSUE TITLE
The auto-detection of assests on a rust release seems to have a few issues with assets

### ISSUE THREAD

**Issue by sheldonhull** [aquaproj/registry-tool#135]
## Reproduce

- Run `aqua-registry scaffold dbrgn/tealdeer`

> Note: I'm submitting a fixed registry entry to the registry, so this is just informational on the auto setup of a package.

See the pointy fingers for the issues I specifically noticed on the defaults: 😄
 
```yaml
packages:
  - type: github_release
    repo_owner: dbrgn
    repo_name: tealdeer
    asset: tealdeer-{{.OS}}-{{.Arch}}-musl
    format: raw
    description: A very fast implementation of tldr in Rust
    replacements:
      amd64: x86_64
      darwin: macos
    overrides:
      - goos: darwin
        asset: tealdeer-{{.OS}}-{{.Arch}}
      - goos: windows
        asset: tealdeer-{{.OS}}-{{.Arch}}-msvc
    supported_envs: # 👈🏻👈🏻👈🏻👈🏻👈🏻  This is a bit confusing. All darwin architecture, else only x86 on everything else, is that what this means? As a newer contributor that's a little confusing.
      - darwin
      - amd64
    rosetta2: true
    checksum:
      type: github_release
      asset: tealdeer-windows-x86_64-msvc.exe.sha256 # 👈🏻👈🏻👈🏻👈🏻👈🏻 
      file_format: regexp
      algorithm: sha256
      pattern:
        checksum: "^(\\b[A-Fa-f0-9]{64}\\b)"
        file: "^\\b[A-Fa-f0-9]{64}\\b\\s+(\\S+)$"

```

**Comment by suzuki-shunsuke** [comment]
I'm sorry but I couldn't find this issue because the notification setting isn't good.

<img width="544" alt="image" src="https://user-images.githubusercontent.com/13323303/215996881-049e10e2-649a-4d38-82a7-6db4ed17fe1d.png">

I changed to `All activetity`.

**Comment by suzuki-shunsuke** [comment]
```
    supported_envs: # 👈🏻👈🏻👈🏻👈🏻👈🏻  This is a bit confusing. All darwin architecture, else only x86 on everything else, is that what this means? As a newer contributor that's a little confusing.
      - darwin
      - amd64
```

Hmm. The document of [supported_envs](https://aquaproj.github.io/docs/reference/registry-config/supported-envs/) is not good...
I'll fix it.

aqua supports the following OS and CPU architecture. https://github.com/aquaproj/aqua-registry/blob/main/CONTRIBUTING.md#supported-os-and-cpu-architecture

If `supported_envs` isn't set, aqua always tries to install a package.
But if `supported_envs` is set, aqua tries to install the package only if the current OS and CPU architecture match with `supported_envs`.

`supported_envs` is `OR` condition, which means `["darwin", "amd64"]` matches "darwin" or "amd64".

- darwin
  - darwin/amd64
  - darwin/arm64
- amd64
  - darwin/amd64
  - linux/amd64
  - windows/amd64

--

- darwin/amd64
- darwin/arm64
- linux/amd64
- windows/amd64

**Comment by suzuki-shunsuke** [comment]
```
      asset: tealdeer-windows-x86_64-msvc.exe.sha256 # 👈🏻👈🏻👈🏻👈🏻👈🏻 
```

asset means GitHub Releases' asset name of a checksum file.

e.g. [tealdeer-windows-x86_64-msvc.exe.sha256](https://github.com/dbrgn/tealdeer/releases/download/v1.6.1/tealdeer-windows-x86_64-msvc.exe.sha256)

`aqua-registry scaffold` command isn't perfect, so sometimes this command generates wrong configuration.
Please fix code yourself. 🙏

**Comment by suzuki-shunsuke** [comment]
I close this issue.
There are some related issues.

- https://github.com/aquaproj/aqua/issues/1667
- https://github.com/aquaproj/aqua/issues/1711
- https://github.com/aquaproj/aqua/issues/1668

https://github.com/aquaproj/aqua/issues/1668 would be released in v1.36.0 and I think this is useful to develop aqua Registry.
